File: Definitions/Game_Definitions/classes/SetAttackHitTypeEffector.lua
---@meta
---@diagnostic disable

---@class SetAttackHitTypeEffector : ModifyAttackEffector
---@field hitType gameuiHitType
SetAttackHitTypeEffector = {}

---@return SetAttackHitTypeEffector
function SetAttackHitTypeEffector.new() return end

---@param props table
---@return SetAttackHitTypeEffector
function SetAttackHitTypeEffector.new(props) return end

---@param record TweakDBID|string
---@param parentRecord TweakDBID|string
function SetAttackHitTypeEffector:Initialize(record, parentRecord) return end

---@param owner gameObject
function SetAttackHitTypeEffector:RepeatedAction(owner) return end

File: DecompiledGameScripts/core/gameplay/effectors/damage/setAttackHitTypeEffector.swift

public class SetAttackHitTypeEffector extends ModifyAttackEffector {

  public let m_hitType: gameuiHitType;

  protected func Initialize(record: TweakDBID, game: GameInstance, parentRecord: TweakDBID) -> Void {
    this.m_hitType = IntEnum<gameuiHitType>(Cast<Int32>(EnumValueFromName(n"gameuiHitType", TweakDBInterface.GetSetAttackHitTypeEffectorRecord(record).HitType())));
  }

  protected func RepeatedAction(owner: ref<GameObject>) -> Void {
    let hitEvent: ref<gameHitEvent> = this.GetHitEvent();
    if !IsDefined(hitEvent) {
      return;
    };
    hitEvent.attackData.SetHitType(this.m_hitType);
  }
}
